# Análise de Dados de Vendas

In [151]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go

In [152]:
vendas = pd.read_csv("Electronic_sales.csv").set_index("Purchase Date")
vendas.index = pd.to_datetime(vendas.index)
vendas.head(5)

Customer ID  Age Gender Loyalty Member Product Type      SKU  \
Purchase Date                                                                 
2024-03-20            1000   53   Male             No   Smartphone  SKU1004   
2024-04-20            1000   53   Male             No       Tablet  SKU1002   
2023-10-17            1002   41   Male             No       Laptop  SKU1005   
2024-08-09            1002   41   Male            Yes   Smartphone  SKU1004   
2024-05-21            1003   75   Male            Yes   Smartphone  SKU1001   

               Rating Order Status Payment Method  Total Price  Unit Price  \
Purchase Date                                                                
2024-03-20          2    Cancelled    Credit Card      5538.33      791.19   
2024-04-20          3    Completed         Paypal       741.09      247.03   
2023-10-17          3    Completed    Credit Card      1855.84      463.96   
2024-08-09          2    Completed           Cash      3164.76      791.19   
2024-05-21          5    Completed           Cash        41.50       20.75   

               Quantity Shipping Type              Add-ons Purchased  \
Purchase Date                                                          
2024-03-20            7      Standard  Accessory,Accessory,Accessory   
2024-04-20            3     Overnight                   Impulse Item   
2023-10-17            4       Express                            NaN   
2024-08-09            4     Overnight      Impulse Item,Impulse Item   
2024-05-21            2       Express                      Accessory   

               Add-on Total  
Purchase Date                
2024-03-20            40.21  
2024-04-20            26.09  
2023-10-17             0.00  
2024-08-09            60.16  
2024-05-21            35.56

In [153]:
vendas_concluidas = vendas[vendas["Order Status"]== "Completed"]
vendas_concluidas

Customer ID  Age  Gender Loyalty Member Product Type      SKU  \
Purchase Date                                                                  
2024-04-20            1000   53    Male             No       Tablet  SKU1002   
2023-10-17            1002   41    Male             No       Laptop  SKU1005   
2024-08-09            1002   41    Male            Yes   Smartphone  SKU1004   
2024-05-21            1003   75    Male            Yes   Smartphone  SKU1001   
2024-05-26            1004   41  Female             No   Smartphone  SKU1001   
...                    ...  ...     ...            ...          ...      ...   
2024-08-09           19995   69  Female            Yes       Laptop   LTP123   
2024-04-20           19996   27  Female             No   Smartphone   SMP234   
2024-06-15           19996   27  Female             No   Smartphone   SMP234   
2024-08-26           19996   27  Female             No   Headphones   HDP456   
2024-01-29           19998   27     NaN            Yes       Laptop   LTP123   

               Rating Order Status Payment Method  Total Price  Unit Price  \
Purchase Date                                                                
2024-04-20          3    Completed         Paypal       741.09      247.03   
2023-10-17          3    Completed    Credit Card      1855.84      463.96   
2024-08-09          2    Completed           Cash      3164.76      791.19   
2024-05-21          5    Completed           Cash        41.50       20.75   
2024-05-26          5    Completed    Credit Card        83.00       20.75   
...               ...          ...            ...          ...         ...   
2024-08-09          3    Completed    Credit Card      5394.56      674.32   
2024-04-20          5    Completed    Credit Card      3419.04     1139.68   
2024-06-15          4    Completed  Bank Transfer      6838.08     1139.68   
2024-08-26          4    Completed  Bank Transfer      1805.90      361.18   
2024-01-29          4    Completed  Bank Transfer       674.32      674.32   

               Quantity Shipping Type  \
Purchase Date                           
2024-04-20            3     Overnight   
2023-10-17            4       Express   
2024-08-09            4     Overnight   
2024-05-21            2       Express   
2024-05-26            4      Standard   
...                 ...           ...   
2024-08-09            8      Same Day   
2024-04-20            3      Same Day   
2024-06-15            6     Expedited   
2024-08-26            5      Standard   
2024-01-29            1     Expedited   

                                        Add-ons Purchased  Add-on Total  
Purchase Date                                                            
2024-04-20                                   Impulse Item         26.09  
2023-10-17                                            NaN          0.00  
2024-08-09                      Impulse Item,Impulse Item         60.16  
2024-05-21                                      Accessory         35.56  
2024-05-26                         Impulse Item,Accessory         65.78  
...                                                   ...           ...  
2024-08-09                                            NaN          0.00  
2024-04-20                                            NaN          0.00  
2024-06-15                                            NaN          0.00  
2024-08-26     Impulse Item, Extended Warranty, Accessory        198.98  
2024-01-29                                            NaN          0.00  

[13432 rows x 15 columns]

In [154]:
vendas_canceladas = vendas[vendas["Order Status"]!= "Completed"]
vendas_canceladas

Customer ID  Age  Gender Loyalty Member Product Type      SKU  \
Purchase Date                                                                  
2024-03-20            1000   53    Male             No   Smartphone  SKU1004   
2023-10-03            1006   24    Male             No   Smartphone  SKU1004   
2023-11-07            1007   35    Male             No   Smartphone  SKU1004   
2024-03-18            1013   42  Female             No   Smartphone  SKU1004   
2024-01-02            1014   73  Female             No   Smartwatch  SKU1003   
...                    ...  ...     ...            ...          ...      ...   
2024-06-12           19994   64    Male             No       Tablet   TBL345   
2024-01-24           19996   27  Female            Yes   Smartphone   SMP234   
2024-04-11           19996   27  Female             No   Smartphone   SMP234   
2024-07-18           19996   27  Female            Yes       Laptop   LTP123   
2024-01-06           19997   27    Male             No   Headphones   HDP456   

               Rating Order Status Payment Method  Total Price  Unit Price  \
Purchase Date                                                                
2024-03-20          2    Cancelled    Credit Card      5538.33      791.19   
2023-10-03          2    Cancelled     Debit Card      5538.33      791.19   
2023-11-07          2    Cancelled    Credit Card      7120.71      791.19   
2024-03-18          2    Cancelled         Paypal      5538.33      791.19   
2024-01-02          3    Cancelled    Credit Card      5068.98      844.83   
...               ...          ...            ...          ...         ...   
2024-06-12          4    Cancelled  Bank Transfer      5504.87      786.41   
2024-01-24          4    Cancelled         PayPal      4558.72     1139.68   
2024-04-11          1    Cancelled         PayPal      7977.76     1139.68   
2024-07-18          4    Cancelled    Credit Card      2697.28      674.32   
2024-01-06          1    Cancelled  Bank Transfer      2528.26      361.18   

               Quantity Shipping Type  \
Purchase Date                           
2024-03-20            7      Standard   
2023-10-03            7      Standard   
2023-11-07            9     Overnight   
2024-03-18            7     Overnight   
2024-01-02            6      Standard   
...                 ...           ...   
2024-06-12            7     Expedited   
2024-01-24            4      Same Day   
2024-04-11            7     Expedited   
2024-07-18            4      Standard   
2024-01-06            7     Expedited   

                                               Add-ons Purchased  Add-on Total  
Purchase Date                                                                   
2024-03-20                         Accessory,Accessory,Accessory         40.21  
2023-10-03                                          Impulse Item         43.05  
2023-11-07                           Accessory,Extended Warranty         55.48  
2024-03-18                      Accessory,Impulse Item,Accessory         57.81  
2024-01-02                   Extended Warranty,Extended Warranty         69.31  
...                                                          ...           ...  
2024-06-12                          Accessory, Extended Warranty         93.17  
2024-01-24                                     Extended Warranty         65.94  
2024-04-11     Extended Warranty, Extended Warranty, Impulse ...        167.20  
2024-07-18                                                   NaN          0.00  
2024-01-06                          Extended Warranty, Accessory        101.34  

[6568 rows x 15 columns]

## Análise das Vendas concretizadas

### Vendas Mensais

In [155]:
#Agrupamento das Vendas por Mês
vendas_mes = vendas_concluidas.resample("MS")[["Total Price"]].sum()
vendas_mes

Total Price
Purchase Date             
2023-09-01       283635.75
2023-10-01      1555325.86
2023-11-01      1390116.50
2023-12-01      1310022.87
2024-01-01      4516277.81
2024-02-01      3897250.16
2024-03-01      4226126.65
2024-04-01      4300323.69
2024-05-01      4462915.20
2024-06-01      4465994.71
2024-07-01      4460946.48
2024-08-01      4378433.10
2024-09-01      3382246.79

In [156]:
fig = px.bar(vendas_mes, y=vendas_mes["Total Price"].round(2),
             title="Vendas tiveram um grande salto no primeiro mês de 2024",
             color="Total Price", color_continuous_scale="Blues")

fig.update_layout(xaxis_title="Mês", yaxis_title="Total em Vendas(em milhões)", template="plotly_dark")

fig.update_traces(text=vendas_mes["Total Price"].apply(lambda x: f"${x / 1e6:,.2f}M"), textposition="inside",
                  hovertemplate="Vendas do Mês: $%{text}<br>Mês: %{x} ")

In [157]:
total_vendas = vendas_concluidas["Total Price"].sum()
print(f" Total em vendas concluídas: {total_vendas:.2f}")

 Total em vendas concluídas: 42629615.57


In [158]:
fig_total = go.Figure(go.Indicator(mode="delta+number+gauge",
                                  value=total_vendas.round(2),
                                  title="Total em Vendas nos últimos 12 meses",
                                  gauge= {"axis": {"range": [0,45000000]} }))
fig_total.add_annotation(x=0.5, y=-0.2, text="Vendas Ultrapassaram os 42 milhões impulsionadas pela forte alta nas vendas apartir de janeiro",
                        showarrow=False, font=dict(size=20, color="green"))
fig_total.show()

### Crescimento das Vendas

In [159]:
#Cálculo da Variação Percentual das Vendas
vendas_mes["crescim_perc"] = vendas_mes.pct_change().fillna(0) *100
vendas_mes

Total Price  crescim_perc
Purchase Date                           
2023-09-01       283635.75      0.000000
2023-10-01      1555325.86    448.353252
2023-11-01      1390116.50    -10.622170
2023-12-01      1310022.87     -5.761649
2024-01-01      4516277.81    244.748013
2024-02-01      3897250.16    -13.706589
2024-03-01      4226126.65      8.438681
2024-04-01      4300323.69      1.755675
2024-05-01      4462915.20      3.780913
2024-06-01      4465994.71      0.069002
2024-07-01      4460946.48     -0.113037
2024-08-01      4378433.10     -1.849683
2024-09-01      3382246.79    -22.752119

In [160]:
fig2 = px.line(vendas_mes, vendas_mes.index, vendas_mes["crescim_perc"] ,
               title="Vendas tiveram um pico de crescimento em novembro e janeiro" )

fig2.update_layout(xaxis_title="Mês", yaxis_title="Crescimento Percentual das Vendas", template="plotly_dark")
fig2.show()

Após fevereiro as vendas se estabilizaram mantendo um bom desempenho geral, mas isso só foi possível graças ao enorme salto nas vendas no mês de novembro
que cresceram mais de 400%

In [161]:
#Cálculo do acúmulo das Vendas utilizando as Vendas Mensais Consolidadas
vendas_acumuladas = vendas_mes[["Total Price"]].cumsum().fillna(0)
vendas_acumuladas = vendas_acumuladas.round(2)
vendas_acumuladas

Total Price
Purchase Date             
2023-09-01       283635.75
2023-10-01      1838961.61
2023-11-01      3229078.11
2023-12-01      4539100.98
2024-01-01      9055378.79
2024-02-01     12952628.95
2024-03-01     17178755.60
2024-04-01     21479079.29
2024-05-01     25941994.49
2024-06-01     30407989.20
2024-07-01     34868935.68
2024-08-01     39247368.78
2024-09-01     42629615.57

In [162]:
fig3 = px.bar(vendas_acumuladas,x=vendas_acumuladas.index, y=vendas_acumuladas["Total Price"],
              title="Veja como as Vendas se Acumulam cada vez mais rápido e atingem níveis significativos ao Longo do Tempo")
              
fig3.update_layout(yaxis_title="Vendas Totalizadas", xaxis_title="Mês", template="plotly_dark")
fig3.update_traces(textposition="outside", hovertemplate="Valor Acumulado até o mês atual: $%{y}<br>Mês: %{x}")
fig3.show()

### Ticket Médio 

In [163]:
ticket_medio = vendas_concluidas.groupby("Product Type")[["Total Price"]].mean()
ticket_medio["Sales Count"] = vendas_concluidas.groupby("Product Type")[["Product Type"]].count()
ticket_medio["Total Price"] = ticket_medio["Total Price"].round(2)
ticket_medio

Total Price  Sales Count
Product Type                          
Headphones        2009.30         1361
Laptop            3114.63         2686
Smartphone        3598.36         4004
Smartwatch        3565.47         2636
Tablet            2813.35         2745

O ticket médio da categoria Smartphone e Smartwatch são bem parecidos porém o que explica a diferença significativa no total de vendas entre as categorias é a alta procura
por smartphones 

In [164]:
print("A categoria Smartphone tem um número de vendas superior a Smartwatch em: {:.2f}%".format(((4004  - 2636) / 2636 )*100))

A categoria Smartphone tem um número de vendas superior a Smartwatch em: 51.90%


In [165]:
print(f"Total em Vendas por Categoria \n {ticket_medio['Total Price'] * ticket_medio['Sales Count']}")

Total em Vendas por Categoria 
 Product Type
Headphones     2734657.30
Laptop         8365896.18
Smartphone    14407833.44
Smartwatch     9398578.92
Tablet         7722645.75
dtype: float64


In [166]:
fig4 = px.bar(data_frame=ticket_medio,x="Total Price", y=ticket_medio.index, title="Ticket Médio por Categoria de Produto", color="Total Price" , color_continuous_scale="Blues")
fig4.update_traces(text=ticket_medio['Total Price'].apply(lambda x: f"${x}"), textposition='inside', hovertemplate="Ticket Médio: $%{text}<br>Categoria: %{y}")

fig4.update_layout(yaxis_title="Categoria do Produto", xaxis_title="Ticket Médio (em $)", template="plotly_dark")

fig4.show()

In [167]:
ticket_medio_mes = vendas_concluidas[["Total Price"]].resample("MS").mean()
ticket_medio_mes

Total Price
Purchase Date             
2023-09-01     2287.385081
2023-10-01     2537.236313
2023-11-01     2588.671322
2023-12-01     2495.281657
2024-01-01     3228.218592
2024-02-01     3176.242999
2024-03-01     3194.351209
2024-04-01     3349.161752
2024-05-01     3255.226258
2024-06-01     3310.596523
2024-07-01     3289.783540
2024-08-01     3309.473243
2024-09-01     3378.867922

In [168]:
fig5 = px.bar(ticket_medio_mes, ticket_medio_mes.index, ticket_medio_mes["Total Price"], 
              title="O ticket Médio Mensal teve um aumento gradativo à medida que as vendas totais aumentaram", color_continuous_scale="Blues")
fig5.update_layout(xaxis_title="Mês" , yaxis_title="Valor do Ticket", template="plotly_dark")
fig5.update_traces(text=ticket_medio_mes["Total Price"].apply(lambda x: f"${x :,.2f}"), textposition="inside", hovertemplate="Ticket Médio: %{y}<br>Mês: %{x}")
fig5.show()

O Ticket médio teve um aumento considerável apartir do primeiro mês de 2024 o que ajuda a explicar o aumento nas vendas em 2024

### Vendas por Categoria

In [169]:
vendas_categoria = vendas_concluidas.groupby("Product Type")[["Total Price"]].sum()
vendas_categoria

Total Price
Product Type             
Headphones     2734651.00
Laptop         8365905.25
Smartphone    14407835.84
Smartwatch     9398591.23
Tablet         7722632.25

In [170]:
fig4 = px.bar(vendas_categoria, y=vendas_categoria.index, x=vendas_categoria["Total Price"],
               title="Devido à alta procura a Categoria Smartphone se destaca no ranking de Vendas", 
                 color=vendas_categoria["Total Price"], color_continuous_scale="Blues")

fig4.update_layout(xaxis_title="Total em Vendas(em milhões)",
                    yaxis_title="Categorias dos Produtos", template="plotly_dark+presentation")

fig4.update_traces(text=vendas_categoria["Total Price"].apply(lambda x: f"${x / 1e6:,.2f}M"),
                    textposition="inside", hovertemplate="Vendas Totais da categoria: %{y}<br> Valor das Vendas: %{x}")

A categoria Smartphone acumula 14 milhões em vendas seguida por Smartwach com quase 10 milhões. Tablet e Laptops próximos à 8 milhões
brigam pelo terceiro lugar nas Vendas.

### Valor Vitalício Médio e Retenção de Clientes

In [171]:
clv = vendas_concluidas.groupby('Customer ID')['Total Price'].sum().mean()
print(f'O Valor Vitalício Médio do Cliente é de: {clv:.2f}')

O Valor Vitalício Médio do Cliente é de: 4503.45


O valor acima é o valor médio que um cliente gasta em toda sua história com a empresa

In [172]:
#Cálculo da Taxa de Retenção de Clientes
cliente_retorno = vendas_concluidas.groupby("Customer ID")["Total Price"].nunique()
taxa_retorno = cliente_retorno[cliente_retorno >1 ].count() / cliente_retorno.count()
print("Taxa de Retorno de Clientes: {:.2f}% ".format(taxa_retorno* 100))

Taxa de Retorno de Clientes: 32.16% 


In [173]:
#ID de alguns Clientes com poucas compras
clientes = 15
print("%s Clientes com menos compras" % clientes)
print(cliente_retorno.sort_values(ascending=True)[:clientes])

15 Clientes com menos compras
Customer ID
1000     1
12624    1
12622    1
12621    1
12619    1
12611    1
12609    1
12608    1
12606    1
12605    1
12603    1
12599    1
12598    1
12597    1
12595    1
Name: Total Price, dtype: int64


In [174]:
fig = go.Figure(go.Indicator(
        mode = "gauge+number",
        value = taxa_retorno*100,
        title = {"text": f"A taxa de retenção de Clientes é de apenas {taxa_retorno*100:.2f}%"},
        align = "center",
        gauge = {"axis": {"range": [0, 100]}, 
                "bar": {"color": "red"}},
                        
))
fig.add_annotation(x=0.5, y=-0.2, text="Apenas 3 em cada 10 clientes voltaram a comprar", 
                  showarrow=False, font=dict(size=19, color="red"))
fig.show()                

## Análise das Vendas Canceladas

A seguir analisaremos algumas visualizações para identificar valores e padrões ocultos que podem ser úteis para tomada de decisão 
como definição de estratégias de marketing, por exemplo

In [175]:
#Cálculo da taxa de Cancelamento
taxa_cancelamento = vendas_canceladas.shape[0] / (vendas_canceladas.shape[0] + vendas_concluidas.shape[0])
print("As Vendas Canceladas representam: {:.2f}% do Total de Vendas ".format(taxa_cancelamento *100))

As Vendas Canceladas representam: 32.84% do Total de Vendas 


### Taxas de Cancelamento

In [176]:
#Visualização Gráfica
fig_canc = go.Figure(go.Indicator(
        mode = "gauge+number",
        value = taxa_cancelamento*100,
        title = {"text": f"A taxa de Cancelamento é de: {taxa_cancelamento*100:.2f}%"},
        align = "left",
        gauge = {"axis": {"range": [0, 100]}, 
                "bar": {"color": "red"}},
                        
))
fig_canc.add_annotation(x=0.5, y=-0.3, text="3 a cada 10 vendas foram canceladas. Vale a pena conferir os feedbacks de possíveis clientes insatisfeitos", 
                  showarrow=False, font=dict(size=19, color="red"))
fig_canc.show()                

In [177]:
#Cálculo dos Clientes com mais cancelamentos
numero_de_clientes = 10
taxa_cliente = vendas_canceladas.groupby("Customer ID")[["Total Price"]].nunique()
taxa_cliente = taxa_cliente["Total Price"].sort_values(ascending=False)
taxa_cliente = taxa_cliente.head(numero_de_clientes)
taxa_cliente

Customer ID
13230    5
12716    4
5442     4
15577    4
17773    4
13792    4
1244     4
17303    4
7318     3
1359     3
Name: Total Price, dtype: int64

In [186]:
fig_canc3 = px.bar(taxa_cliente,taxa_cliente.index.astype(str),taxa_cliente, color="Total Price",
                   title=f"Estes são os {numero_de_clientes} clientes com o maior número de cancelamentos", color_continuous_scale="Reds")

fig_canc3.update_layout(yaxis_title="Número de Cancelamentos", 
                        xaxis_title="ID do Cliente", template="plotly_dark")

fig_canc3.update_traces(text=taxa_cliente.values, textposition="none",
                         hovertemplate="Total de Vendas Canceladas: %{text}<br>ID do Clinte %{x} ")

fig_canc3.show()

Vale a pena coletar o feedback desses clientes para identificar possíveis insatisfações em comum, como problemas com a plataforma ou com o atendimento por exemplo

### Cancelamento por Categoria

In [187]:
cancelamento_categoria = vendas_canceladas.groupby("Product Type")[["Product Type"]].count()
cancelamento_categoria

Product Type
Product Type              
Headphones             650
Laptop                1287
Smartphone            1974
Smartwatch            1298
Tablet                1359

In [188]:
fig_canc3 = px.bar(cancelamento_categoria,cancelamento_categoria["Product Type"],cancelamento_categoria.index, color="Product Type",
                   title="As vendas canceladas por categoria seguem a tendência das Vendas", color_continuous_scale="Reds")
fig_canc3.update_layout(yaxis_title="Categoria", xaxis_title="Total de Vendas Canceladas", template="plotly_dark+gridon")
fig_canc3.update_traces(text=cancelamento_categoria["Product Type"], textposition="inside", hovertemplate="Total de Vendas Canceladas: %{text}<br>Categoria do Produto %{y} ")
fig_canc3.show()

### Cancelamento por Mês

In [189]:
cancelamento_mes = vendas_canceladas.resample("MS")[["Order Status"]].count()
cancelamento_mes

Order Status
Purchase Date              
2023-09-01               66
2023-10-01              298
2023-11-01              260
2023-12-01              275
2024-01-01              650
2024-02-01              591
2024-03-01              634
2024-04-01              648
2024-05-01              677
2024-06-01              650
2024-07-01              633
2024-08-01              695
2024-09-01              491

In [190]:
fig_canc4 = px.bar(cancelamento_mes,y=cancelamento_mes["Order Status"],x=cancelamento_mes.index,
                   title="Os cancelamentos seguem a tendência ascendente das Vendas", color="Order Status", color_continuous_scale="Reds", template="plotly_white")
fig_canc4.update_layout(xaxis_title="Mês", yaxis_title="Total de Vendas Canceladas")
fig_canc4.update_traces(text=cancelamento_mes["Order Status"], textposition="inside", hovertemplate="Total de Vendas Canceladas: %{text}<br>Data %{x} ")
fig_canc4.show()

A tendência de cancelamentos acompanha a tendência de Crescimento das Vendas, mas com algumas ressalvas: O mês de Agosto apesar de não ser o mês com maior 
número de Vendas é o mês com maior número de Cancelamentos, e o mês de Janeiro que foi o melhor mês de Vendas tem um número aceitável de Cancelamentos.
Vale a pena checar os fatores que podem ter influenciado esse comportamento. 

### Cancelamento das Vendas no Programa de Fidelidade

In [191]:
cancelamentos_membros = vendas_canceladas.groupby("Loyalty Member")[["Loyalty Member"]].size()
cancelamentos_membros

Loyalty Member
No     5145
Yes    1423
dtype: int64

In [192]:
porc_cancel = cancelamentos_membros.iloc[1] / (cancelamentos_membros.iloc[0] + cancelamentos_membros.iloc[1])
print(f"Os clientes que fazem parte do programa representam apenas {porc_cancel*100:.2f}% dos cancelamentos totais")

Os clientes que fazem parte do programa representam apenas 21.67% dos cancelamentos totais


In [193]:
#Criação do gráfico com título e escala de cores
fig_canc5 = px.bar(cancelamentos_membros, cancelamentos_membros.index, cancelamentos_membros.values,
                   color=cancelamentos_membros.values, color_continuous_scale="Reds",
                   title="Clientes que fazem parte do programa de fidelização tem uma taxa de Cancelamento consideravelmente menor")

#Adição de texto nos eixos e tema para visualização mais agradável
fig_canc5.update_layout(xaxis_title="Cliente no Programa de Fidelização", 
                        yaxis_title="Total de Cancelamentos", template="plotly_white")

#Atualização dos textos das barras e da interação
fig_canc5.update_traces(text=cancelamentos_membros.values, textposition="inside", 
                        hovertemplate="Cliente na Fidelização: %{x}<br>Cancelamentos: %{text} ", marker=dict(line=dict(color="black", width=1.75)))

#Configuração da seta, anotacões adicionais e respectivas coordenadas 
fig_canc5.add_annotation(x=1.0, y=cancelamentos_membros.values[1], text=f"Isso representa apenas {porc_cancel*100:.2f}% dos clientes ", ax=0.0, ay=-80, arrowhead=3,
                         showarrow=True, arrowsize=2, font=dict(size=14, color="green"))

#Visualização do Gráfico
fig_canc5.show()

Pode ser uma boa estratégia investir na fidelização dos clientes pois os clientes que fazem parte do programa de fidelização tendem a ser mais fiéis à empresa,  
com uma taxa de cancelamento consideravelmente menor, representando apenas 21% dos cancelamentos totais, como mostrado no gráfico acima.

### Conclusão

A empresa de um modo geral apresenta um bom volume de vendas mas com espaço para melhorias. Como já mostrado nos gráficos acima uma boa estratégia  pode ser a fidelização de clientes através de programas, fidelização, bônus, cupons, vantagens exclusivas, etc.
Coletar o feedback de possíveis clientes insatisfeitos também pode identificar áreas de melhoria e insatisfações em comum dos clientes.. E porfim clientes com poucas compras podem receber uma abordagem personalizada do departamento de marketing para 
insentivar o re-engajamento e novas compras dos clientes.

### Futura Melhorias

Melhoria que pode ser implementada em versão posterior: Adição de um modelo de machine learning para identificar clientes que podem cancelar suas compras